spec.bs

```diff
@@ -8349,6 +8630,101 @@ scoring signals requests=] |entriesToBatch|:
 
 <div algorithm>
 
+To <dfn>build trusted key value scoring signals request body</dfn> given a non-empty [=list=] of
+[=trusted scoring signals requests=] |entriesToBatch|:
+
+1. Let |firstRequest| be |entriesToBatch|[0].
+1. Let |compressionGroups| be an empty [=list=], whose values are [=maps=].
+1. Let |compressionGroupMap| be an empty [=map=], whose [=map/keys=] are integers and keys are [=maps=].
+1. Let |compressionIdMap| be an empty [=map=], whose [=map/keys=] are [=tuples=] of ([=origin=],
+  [=origin=]) and [=map/values=] are integers.
+1. Let |renderUrlIdMap| be an empty [=map=], whose [=map/keys=] are [=URLs=] and [=map/values=] are [=tuples=]
+  of (interger, integer).
+1. Let |nextCompressionGroupId| be 0.
+1. [=map/For each=] |request| of |entriesToBatch|:
+  1. Let |partitionId| be an integer.
+  1. Set |joiningOrigin| to |group|'s [=interest group/joining origin=].
+  1. Let |ownerOrigin| be |request|'s [=trusted scoring signals request/owner origin=].
+  1. Let |mapKey| be a tuple of |ownerOrigin| and |joiningOrigin|.
+  1. If |compressionIdMap| does not [=map/contain=] |mapKey|:
+    1. Set |compressionIdMap|[|mapKey|] to |nextCompressionGroupId|.
+    1. Increase |nextCompressionGroupId| by 1.
+  1. Let |compressionGroupId| be |compressionIdMap|[|mapKey|].
+  1. If |compressionGroupMap| does not [=map/contain=] |compressionGroupId|:
+    1. Let |compressionGroupMap|[|compressionGroupId|] be an empty [=map=], whose [=map/keys=] are integers
+      and [=map/values=] are [=maps=].
+  1. Set |partitionId| to [=list/size=] of |compressionGroupMap|[|compressionGroupId|].
+  1. Set |renderUrlIdMap|[|request|'s [=trusted scoring signals request/render URL=]] to [=tuple=]
+    of |compressionGroupId| and |partitionId|.
+  1. Let |partition| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are [=strings=] or
+      [=list=] of [=strings=].
+  1. Set |partition|["id"] to |partitionId|.
+  1. Let |namespace| be an empty [=map=], whose [=map/keys=] are [=strings=] and [=map/values=] are [=strings=] or
+      [=list=] of [=strings=].
+  1. Set |namespace|["render_url"] to [|request|'s [=trusted scoring signals request/render URL=]].

```

Comment:
it seems |namespace| is more like a [struct](https://infra.spec.whatwg.org/#struct) with specific keys, instead of a map. You can define a new structure type for namespace, with short explainations of each field, and I think that'll be clearer as well.

Reply:
This `namespace` is defined as a map in IETF spec, which is consistent with the definition here.